nix/overlay.nix

```diff
@@ -22,7 +22,25 @@ rec {
 
   # Override "nixpkgs" circt with "nixpkgs-for-circt".
   # To update the "nixpkgs-for-circt" input, run `nix flake lock --update-input nixpkgs-for-circt`.
-  circt = self.inputs.nixpkgs-for-circt.legacyPackages."${final.system}".circt;
+  circt = self.inputs.nixpkgs-for-circt.legacyPackages."${final.system}".circt.overrideAttrs (prevAttrs: {
```

Comment:
Here introduce some unsafe codes, in the future we need to consider ask T1 to depends on zaozi with its Scala rewritten firtool to align the toolchain.